cpp/include/algorithms.hpp

```diff
@@ -1187,5 +1187,33 @@ extract_ego(raft::handle_t const &handle,
             vertex_t *source_vertex,
             vertex_t n_subgraphs,
             vertex_t radius);
+
+/**
+ * @brief returns random walks (RW) from starting sources, where each path is of given maximum
+ * length. Uniform distribution is assumed for the random engine.
+ *
+ * @tparam graph_t Type of graph.
+ * @tparam vertex_type Type of vertex identifiers. Needs to be an integral type.
+ * @tparam weight_type Type of edge weights. Needs to be a floating point type.
+ * @param handle RAFT handle object to encapsulate resources (e.g. CUDA stream, communicator, and
+ * handles to various CUDA libraries) to run graph algorithms.
+ * @param graph Graph object to generate RW on.
+ * @param ptr_d_start Device pointer to set of starting vertex indices for the RW.
+ * @param num_paths = number(paths) == d_v_start.size().
+ * @param max_depth maximum length of RWs.
+ * @return std::tuple<device_vec_t<vertex_t>, device_vec_t<weight_t>,
+ * device_vec_t<index_t>> Triplet of coalesced RW paths, with corresponding edge weights for
+ * each, and coresponding path sizes. This is meant to minimize the number of DF's to be passed to
+ * the Python layer.
+ */
+template <typename graph_t, typename index_t>
+std::tuple<rmm::device_uvector<typename graph_t::vertex_type>,
+           rmm::device_uvector<typename graph_t::weight_type>,
+           rmm::device_uvector<index_t>>
+random_walks(raft::handle_t const &handle,
+             graph_t const &graph,
```

Comment:
It works with `graph_view_t`. As you can see `graph_t` is just the local name of a template parameter.

Reply:
Is there really a fundamental reason this shouldn't take ```graph_view_t<vertex_t, edge_t, weight_t, false, multi_gpu> const &graph_view``` like other functions in the experimental space? If not, I think it is better to be consistent with the rest of the functions.

Reply:
That seems reasonable.  The louvain example (which uses `graph_t` - changed to `graph_view_t` in the latest PR) that Andrei may have copied this from does this because the *same* API supports both experimental and legacy graph classes.  Things that only support the experimental graph classes should follow the explicit paradigm, I think.

Reply:
Yes, 2 reasons:
(1) Preparation for the many graph types that we will soon have in our code.
(2) Hiding template parameters that don't need to be made explicit;

I can rename `graph_t<>` as `graph_view_t` but I don't see a reason to expose `edge_t`, `weight_t` separately, when they don't explicitly contribute to the API (they do implicitly, through the graph type...but that's exactly the reason I don't want them explicit).

Also, why explicitly supply the boolean template parameters? By _not_ supplying them, I won't have to `enable_if_t` and provide a separate declaration for this API, when the MNMG implementation will be added.

Reply:
FYI - You probably will somewhere in the call stack need to add an enable_if_t even if you don't supply the boolean template parameters here.  The detail methods you call ultimately need you (at least currently) to do that.

Reply:
Yes, I have the `enable_if_t` in the details. But why propagate it higher in the API, if it can be avoided?